### PR TITLE
Move deploy_trento test code to lib

### PR DESCRIPTION
**Pull request by mpagot**
Move more code from deploy_trento to lib/trento
Create a lib function for the ACR part
Create a lib function for the Trento installation
Move VM creation from the bash script to a new python function More and more unit test for the trento lib
Made all mocks in the trento unit testing private for each test

- Verification run:  http://1c242.qa.suse.de/tests/1348  https://openqaworker15.qa.suse.cz/tests/58226 https://openqaworker15.qa.suse.cz/tests/58227 https://openqaworker15.qa.suse.cz/tests/59353

### CODE REVIEW

**Review by jankohoutek — APPROVED** [review]
LGTM

**Review by jankohoutek — APPROVED** [review]
also LGTM

File: lib/trento.pm [hunk 1]
```diff
@@ -257,25 +259,139 @@ Based on 00.040-trento_vm_server_deploy_azure.sh
...
+    my $cfg_json = 'config_images_gen.json';
+    my @imgs = qw(WEB RUNNER);
+    # take care of TRENTO_VERSION variable too
+    if (get_var("TRENTO_REGISTRY_IMAGE_$imgs[0]") || get_var("TRENTO_REGISTRY_IMAGE_$imgs[1]")) {
+        my $cfg_helper_cmd = TRENTO_SCRIPT_RUN . 'trento_deploy/config_helper.py' .
+          " -o $cfg_json" .
+          " --chart $trento_registry_chart";
+        if (get_var('TRENTO_REGISTRY_CHART_VERSION')) {
+            $cfg_helper_cmd .= ' --chart-version ' . get_var('TRENTO_REGISTRY_CHART_VERSION');
+        }
```
[the remark this answers is not in the document]
Comment (mpagot) [reply lib/trento.pm:304]: The logical condition TRENTO_REGISTRY_IMAGE_WEB or TRENTO_REGISTRY_IMAGE_RUNNERtrigger two events:

1.  result in the definition of the $cfg_helper_cmd but, more importantly, result in the execution of the **config_helper.py** script that result in generation of the **config_images_gen.json** file
2. result in **$trento_registry_chart** content to be replaced with json filename

**Review by lpalovsky — APPROVED** [review]
LGTM

File: lib/trento.pm [hunk 2]
```diff
@@ -257,25 +259,139 @@ Based on 00.040-trento_vm_server_deploy_azure.sh
...
+    push @cmd_list, TRENTO_SCRIPT_RUN . $script_id . '-trento_server_installation_premium_v.sh';
+    push @cmd_list, ('-i', $machine_ip);
+    push @cmd_list, ('-k', SSH_KEY);
+    push @cmd_list, ('-u', VM_USER);
+    if (get_var('TRENTO_REGISTRY_CHART_VERSION')) {
+        push @cmd_list, ('-c', get_var('TRENTO_REGISTRY_CHART_VERSION'));
+    }
+    if (get_var("TRENTO_REGISTRY_IMAGE_$imgs[0]") || get_var("TRENTO_REGISTRY_IMAGE_$imgs[1]")) {
+        push @cmd_list, ('-x', $acr->{'trento_cluster_install'});
+    }
```
[the remark this answers is not in the document]
Comment (mpagot) [reply lib/trento.pm:382]: > Suggestion:
> 
> ```
> push @cmd_list, ('-x', $acr->{'trento_cluster_install'})
>     if (get_var("TRENTO_REGISTRY_IMAGE_$imgs[0]", get_var("TRENTO_REGISTRY_IMAGE_$imgs[1]")));
> ```

Is it a trick to get an OR exploiting **get_var** behavior about argument for default value?

[the remark this answers is not in the document]
Comment (alvarocarvajald) [reply lib/trento.pm:382]: Yes. :)

File: lib/trento.pm (hunk 1 above)
[the remark this answers is not in the document]
Comment (alvarocarvajald) [reply lib/trento.pm:304]: They way I'm reading this is:
```
if (condition1 or condition2):
    cfg_helper_cmd = some_command
    if condition3:
        cfg_helper_cmd = cfg_helper_cmd + some_args
        for img in (condition1, condition2):
            if (img):    # <-- Note that here either condition1 is true, or condition2 is true, or both are true
                cfg_helper_cmd = cfg_helper_cmd + some_other_args
                if (condition4):
                    cfg_helper_cmd = cfg_helper_cmd + some_other_extra_args
                else:
                    cfg_helper_cmd = cfg_helper_cmd + 'latest'
    run cfg_helper_cmd
    upload_logs
    set trento_registry_chart
```
I think the same can be accomplished by:
```
cfg_helper_cmd = some_command
if condition3:
    cfg_helper_cmd = cfg_helper_cmd + some_args
    for img in (condition1, condition2):
        if (img):    # <-- Note that here the block below is not executed when condition1 and condition2 are both false
            cfg_helper_cmd = cfg_helper_cmd + some_other_args
            if (condition4):
                cfg_helper_cmd = cfg_helper_cmd + some_other_extra_args
            else:
                cfg_helper_cmd = cfg_helper_cmd + 'latest'
if (condition1 or condition2):
    run cfg_helper_cmd
    upload_logs
    set trento_registry_chart
```
It's not obvious, but this way you're removing one level of `if` and `for` blocks which can be helpful in the future (there is a limit per perl tidy of 6 IIRC)

**Review by alvarocarvajald — APPROVED** [review]
LGTM

File: lib/trento.pm [hunk 3]
```diff
@@ -257,25 +259,129 @@ Based on 00.040-trento_vm_server_deploy_azure.sh
...
+    enter_cmd "cd $work_dir";
     my $script_id = '00.040';
     my $resource_group = get_resource_group();
     my $machine_name = get_vm_name();
     record_info($script_id);
-    my $vm_image = get_var(TRENTO_VM_IMAGE => 'SUSE:sles-sap-15-sp3-byos:gen2:latest');
+    my $vm_image = get_var('TRENTO_VM_IMAGE', 'SUSE:sles-sap-15-sp3-byos:gen2:latest');
     my $deploy_script_log = "script_$script_id.log.txt";
-    my $cmd = join(' ', TRENTO_SCRIPT_RUN . $script_id . '-trento_vm_server_deploy_azure.sh ',
+    my $cmd = join(' ', TRENTO_SCRIPT_RUN . 'trento_deploy/trento_deploy.py', '--verbose', '00_040',
```
Comment (mpagot) [inline lib/trento.pm:270]: This is important line

**Pull request merged**